src/rgw/rgw_sal_motr.cc

```diff
@@ -1025,8 +1025,12 @@ int MotrBucket::unlink_user(const DoutPrefixProvider* dpp, const rgw_user &bucke
 {
   // Remove the user into the user info index.
   bufferlist bl;
+  string user_info_idx_name;
   string tenant_bkt_name = get_bucket_name(info.bucket.tenant, info.bucket.name);
-  string user_info_idx_name = "motr.rgw.user.info." + bucket_owner.id;
+  if (info.bucket.tenant != "")
+    user_info_idx_name = "motr.rgw.user.info." + info.bucket.tenant + "$" + bucket_owner.id;
+  else
+    user_info_idx_name = "motr.rgw.user.info." + bucket_owner.id;
```

Comment:
We don't need to check if there is a value for tenant or not, if tenant is empty, it will append empty string.

Also, we already have `to_str()` function for prefixing tenant, so just replace `bucket_owner.id` with `bucket_owner.to_str()`

Reply:
Addressed. Thanks @jjxsg